src/redisai.c

```diff
@@ -232,122 +223,128 @@ int RedisAI_TensorSet_RedisCommand(RedisModuleCtx *ctx, RedisModuleString **argv
     return RedisModule_ReplyWithError(ctx, REDISMODULE_ERRORMSG_WRONGTYPE);
   }
 
-  // getting the datatype
-  const char* typestr;
-  AC_GetString(&ac, &typestr, NULL, 0); 
-
+  // get the tensor datatype
+  const char* typestr = RedisModule_StringPtrLen(argv[2], NULL);
   size_t datasize = RAI_TensorDataSizeFromString(typestr);
   if (!datasize){
     return RedisModule_ReplyWithError(ctx, "ERR invalid data type");
   }
 
-  int dims_arg = 3;
-
-  ArgsCursor dac;
```

Comment:
hi there @lantiga, I believe that the usage of AC should be a tradeoff between simplicity/normalization of argument parsing vs performance. 
I believe AC is best suited for parsing arguments on commands with severall options that make unberable or unreadable using strcasecmp and the methods provided by the redis module API. For this specific case I believe we should not use AC since without it we simplify and increase performance. 
another example is a module created by @antirez is https://github.com/antirez/disque-module/blob/master/job.c#L1066 that uses plain strcasecmp. 
imho both AC or the simple way are valid, but for hot path and simple commands we should try to simplify things :)

Reply:
I get it I get it :-)